src/Pages/Login/Login.tsx

```diff
@@ -0,0 +1,234 @@
+import React, { useReducer } from 'react'
+import axios from 'axios'
+import './login.css';
+import { useNavigate, useLocation } from "react-router-dom";
+import { useAuth } from '../../Contexts'
+import { LoginType } from './LoginType';
+import { loginReducer } from "../../Reducers/loginReducer";
+import 'react-notifications-component/dist/theme.css';
+import { store } from 'react-notifications-component';
+
+export function Login() {
+
+    const initialState: LoginType = { loginBottomBorder: "1px solid var(--color-text)", signupBottomBorder: "none", loginCredentials: { username: "", password: "" }, signupCredentials: { name: "", gender: "", email: "", username: "", password: "", confirmPassword: "" } }
+    const [loginState, loginDispatch] = useReducer(loginReducer, initialState);
+    const navigate = useNavigate();
+    const { authDispatch } = useAuth();
+    const { pathname } = useLocation();
+
+    async function loginHandler(data: { username: string, password: string }) {
+        try {
+            if (data.username === "" || data.password === "") {
+                return store.addNotification({
+                    message: 'UserName or password cannot be empty.',
+                    type: "danger",
+                    insert: "top",
+                    container: "top-center",
+                    animationIn: ["animate__animated", "animate__fadeIn"],
+                    animationOut: ["animate__animated", "animate__fadeOut"],
+                    dismiss: {
+                        duration: 3000,
+                        onScreen: false
+                    }
```

Comment:
creating function for this(object which is passed) which returns object (according to passed values in func) will be helpful and reusable